ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/KsqlRestApplication.java

```diff
@@ -273,12 +274,11 @@ public static KsqlRestConfig convertToApiServerConfig(final KsqlRestConfig confi
     this.configurables = requireNonNull(configurables, "configurables");
     this.rocksDBConfigSetterHandler =
         requireNonNull(rocksDBConfigSetterHandler, "rocksDBConfigSetterHandler");
+    this.pullQueryExecutor = requireNonNull(pullQueryExecutor, "pullQueryExecutor");
```

Comment:
how are the other executors created? are we departing from any pattern..

Reply:
Other executors follow a different pattern since they don't have constructors